terraform/treasury_generation_step_function.tf

```diff
@@ -0,0 +1,137 @@
+module "treasury_generation_step_function" {
+  source = "terraform-aws-modules/step-functions/aws"
+
+  name = "generate-treasury-report"
+  definition = jsonencode({
+    "Comment" : "Generate all the files for a treasury report",
+    "StartAt" : "Parallel",
+    "States" : {
+      "Parallel" : {
+        "Type" : "Parallel",
+        "End" : true,
+        "Branches" : [
+          {
+            "StartAt" : "Generate Project 1A File",
+            "States" : {
+              "Generate Project 1A File" : {
+                "Type" : "Task",
+                "Resource" : "arn:aws:states:::lambda:invoke",
+                "OutputPath" : "$.Payload",
+                "Parameters" : {
+                  "Payload.$" : "$",
+                  "FunctionName" : module.lambda_function-treasuryReportGeneration["1A"].lambda_function_arn
```

Comment:
See my previous comment; I think it would be better to inject a "report type" parameter into the payload and have `FunctionName` point to `module.lambda_function-treasuryReportGeneration.lambda_function_arn`.